packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -223,109 +225,235 @@ function makePreparableAnalyticsAction<
 }
 
 export type AnalyticsActionOptions<
-  LegacyStateNeeded extends
-    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
-  StateNeeded extends
-    StateNeededBySearchAnalyticsProvider = StateNeededBySearchAnalyticsProvider,
-  PayloadType extends RelayPayload = RelayPayload,
-> = LegacyAnalyticsOptions<LegacyStateNeeded> &
-  Partial<NextAnalyticsOptions<StateNeeded, PayloadType>>;
+  LegacyStateNeeded extends StateNeededBySearchAnalyticsProvider,
```

Comment:
I used `=` to avoid 'type drilling' (think prop drilling but for types). However, with the factory, this doesn't make much sense anymore because every type and extend comes from the beginning